### PR TITLE
ci: update on-board process

### PR DESCRIPTION
**Pull request by yihau**
- refactor existing pipeline
- add discord role pipeline

it looks like:

Screenshot 2023-05-12 at 2 50 06 AM

### CODE REVIEW

**Comment by yihau** [comment]
has tested it in my another testing organization. will test it again here when it merged

File: .github/workflows/add-discord-role.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+          roleid=$(
+            curl -s -H "Authorization: Bot ${{ secrets.DISCORD_DEVBOT_TOKEN }}" \
+              "https://discord.com/api/guilds/${{ secrets.DISCORD_SOLANA_GUILD_ID }}/roles" | \
+              jq -r '.[] | select(.name=="Contributor") | .id'
+          )
+          if [[ $roleid = "null" ]]; then
+            echo "failed to get role id"
+            exit 1
+          fi
```
Comment (joeaba) [inline .github/workflows/add-discord-role.yml:51]: This is fine, but in the future you can just do jq -e -r to simplify the error handling

`userid=$(
  curl -s -H "Authorization: Bot ${{ secrets.DISCORD_DEVBOT_TOKEN }}" \
    "https://discord.com/api/guilds/${{ secrets.DISCORD_SOLANA_GUILD_ID }}/members/search?limit=1000&query=$username" | \
    jq -e --raw-output --arg discriminator "$discriminator" '.[] | select(.user.discriminator == $discriminator) | .user.id'
) || {
  echo "failed to get user id"
  exit 1
}

roleid=$(
  curl -s -H "Authorization: Bot ${{ secrets.DISCORD_DEVBOT_TOKEN }}" \
    "https://discord.com/api/guilds/${{ secrets.DISCORD_SOLANA_GUILD_ID }}/roles" | \
    jq -e -r '.[] | select(.name=="Contributor") | .id'
) || {
  echo "failed to get role id"
  exit 1
}`

File: .github/workflows/add-discord-role.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            echo "unexpected discord handle"
+            exit 1
+          fi
+
+          username="${BASH_REMATCH[1]}"
+          discriminator="${BASH_REMATCH[2]}"
+
+          userid=$(
+            curl -s -H "Authorization: Bot ${{ secrets.DISCORD_DEVBOT_TOKEN }}" \
+              "https://discord.com/api/guilds/${{ secrets.DISCORD_SOLANA_GUILD_ID }}/members/search?limit=1000&query=$username" | \
```
Comment (joeaba) [inline .github/workflows/add-discord-role.yml:35]: This is fine, but in the future you can just use jq -e -r to simplify the error handling

**Review by joeaba — APPROVED** [review]

**Comment by joeaba** [comment]
Looks goods, the only thing missing is the other roles specific to the level of access they have e.g. Monorepo-Write, etc, but it's fine we can add that later. If you are not aware, beside getting the contributor role they also get a role with the type of access they have

**Comment by yihau** [comment]
ah.. my bad. I don't notice that we have different roles for them 😢. will add them later.

**Pull request merged**